### PR TITLE
GIPF-28: Setup

### PR DESCRIPTION
**Pull request by sulphite**
installed dependencies and set up tsconfig to output to dist folder.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (adrianHards) [inline package.json:1]: odd indentation

File: package.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	"dependencies": {
```
Comment (adrianHards) [inline package.json:2]: odd indentation

Comment (adrianHards) [inline package.json:2]: check indentation; would be good to add linter early to enforce styling rules

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Dependencies
```
Comment (adrianHards) [inline .gitignore:1]: I'd recommend removing unnecessary items from gitignore; only add what's needed for now

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const express = require("express");
+import { createServer } from 'http';
+import { Server } from 'socket.io';
```
Comment (adrianHards) [inline src/index.ts]: wouldn't add socket until we're using it

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "outDir": "dist",
+    "target": "es2017",
```
Comment (adrianHards) [inline tsconfig.json]: did these come by default or were these added? Make sure you know what each does if the latter

**Review by adrianHards — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (sulphite) [reply package.json:2]: strange, looks normal in my code editor, but ok

File: tsconfig.json (hunk 5 above)
Comment (sulphite) [reply tsconfig.json]: we need outdir to tell it where to put the transpiled JS. I've updated target to ES2020 so we can use newer features

**Review by adrianHards — APPROVED** [review]

**Pull request merged**